cppapi/server/dserver.cpp

```diff
@@ -1985,29 +1981,30 @@ void DServer::mcast_event_for_att(std::string &dev_name,std::string &att_name,st
 //
 //------------------------------------------------------------------------------------------------------------------
 
-void DServer::mem_event_par(ServerEventSubscriptionState& _map)
+ServerEventSubscriptionState DServer::get_event_subscription_state()
 {
-	for (size_t i = 0;i < class_list.size();i++)
-	{
-		std::vector<DeviceImpl *> &dev_list = class_list[i]->get_device_list();
-		for (size_t j = 0;j < dev_list.size();j++)
-		{
-			EventSubscriptionStates eve;
-			dev_list[j]->get_device_attr()->get_event_param(eve);
-			dev_list[j]->get_event_param(eve);
+    ServerEventSubscriptionState result{};
 
-			if (eve.size() != 0)
-			{
-				_map.insert(make_pair(dev_list[j]->get_name(),eve));
-			}
-		}
-	}
+    for (const auto& device_class : class_list)
+    {
+        for (const auto& device : device_class->get_device_list())
+        {
+            auto events = device->get_event_subscription_state();
+
+            if (events.has_dev_intr_change_event_clients ||
+                ! events.attribute_events.empty())
+
+            result.emplace(device->get_name(), std::move(events));
```

Comment:
I'm not sure how to read that code.

Should it be

```
            if (events.has_dev_intr_change_event_clients ||
                ! events.attribute_events.empty())
             {
                 result.emplace(device->get_name(), std::move(events));
             }
```
?

Reply:
Thanks! Correected as you suggested.